### PR TITLE
In memory as CLI parameter

### PR DESCRIPTION
**Pull request by reds-heig**
Hi,

This PR enable the possibility of enable or not the `in memory` version.

* The parameter `--in-memory` is now available for users from the CLI.
* Removed from the compilation `*-memory ` binaries.
* Test updated and fixed, because previously the in memory test did not run on the in memory data structure.

Remaining work before merging

* [ ]  Update documentation

If you have any suggestion or questions, please feel free :)

Best,
Julián

### CODE REVIEW

File: test/base.h [hunk 1]
```diff
@@ -54,27 +55,52 @@ void compute(std::string&& filename, std::vector<size_t> homology, std::string o
...
-	compute("../../test/f.flag", {{1ul, 0ul, 0ul}});
-	compute("../../test/d3.flag", {{1ul, 0ul, 2ul}});
-	compute("../../test/d3-allzero.flag", {{1ul, 0ul, 2ul}});
-	compute("../../test/double-d3.flag", {{1, 0, 5}});
-	compute("../../test/double-d3-allzero.flag", {{1, 0, 5}});
-	compute("../../test/d4.flag", {{1, 0, 0, 9}});
-	compute("../../test/d4-allzero.flag", {{1, 0, 0, 9}});
-	compute("../../test/d5.flag", {{1, 0, 0, 0, 44}});
-	compute("../../test/d7.flag", {{1, 0, 0, 0, 0, 0, 1854}});
+void run_all(bool full = false, bool in_memory = false) {
```
Comment (luetge) [inline test/base.h]: How about creating a run_all that depends on a type of flag complex computer just like you did for compute? Then you can call run_all<dfc_memory>(true/false) etc. If you want to hide that complexity you can still crate a wrapper that does the translation of bool into type, but probably not necessary.

**Review by luetge — CHANGES_REQUESTED** [review]
Looks great overall, thanks! Let me know if you agree with my proposal and then we move forward. Sorry for the delay!

File: test/base.h (hunk 1 above)
Comment (MonkeyBreaker) [reply test/base.h]: Nice idea, I didn't think about it but it makes thing much clear :)

**Comment by MonkeyBreaker** [comment]
Don't worry about the delay, you do it when you have time ;)

File: test/base.h (hunk 1 above)
Comment (luetge) [reply test/base.h]: I pushed some commits, would be ready to merge. I hope I did not accidentally prevent some additions that you were still planning, let me know when you want to merge!

Comment (MonkeyBreaker) [reply test/base.h]: Nothing more to add on my side.
When you're good on your side, I think that this PR can be merged :+1:

**Review by luetge — APPROVED** [review]
Thanks a lot!

**Pull request merged**